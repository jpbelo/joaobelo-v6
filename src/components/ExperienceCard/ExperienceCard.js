import React, { Component } from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { space } from 'styled-system'

import TechsList from 'components/TechsList'

const ExperienceCardContainer = styled.div`
  ${space}
  h2 {
    margin-bottom: 10px;
  }
  h3 {
    font-size: 12px;
    margin-bottom: 4px;
  }
  p {
    margin-bottom: 15px;
  }
`

const PositionCardContainer = styled.div`
  border-left: 1px solid #999;
  padding-left: 10px;
`

export default class ExperienceCard extends Component {
  render() {
    const { company, positions, techs } = this.props
    return (
      <ExperienceCardContainer mb={4}>
        <h2>{company}</h2>
        {positions.map((position) => (
          <PositionCardContainer key={position.id}>
            <h3>
              {position.dateStart} to {position.dateEnd}
            </h3>
            <p dangerouslySetInnerHTML={{ __html: position.info }} />
          </PositionCardContainer>
        ))}
        <TechsList techsList={techs} />
      </ExperienceCardContainer>
    )
  }
}

ExperienceCard.propTypes = {
  id: PropTypes.number.isRequired,
  company: PropTypes.string.isRequired,
  positions: PropTypes.arrayOf(
    PropTypes.shape({
      info: PropTypes.string.isRequired,
      dateStart: PropTypes.string.isRequired,
      dateEnd: PropTypes.string.isRequired,
    }),
  ),
  techs: PropTypes.array.isRequired,
}

ExperienceCard.defaultProps = {
  positions: [],
}
