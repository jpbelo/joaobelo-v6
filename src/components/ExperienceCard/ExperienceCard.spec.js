import React from 'react'
import 'jest-styled-components'

import { renderWithTheme } from 'helpers/renderWithTheme'
import ExperienceCard from './ExperienceCard'

const mockProps = {
  id: 4,
  dateStart: '2019-04-19',
  dateEnd: 'present',
  company: 'Waitrose',
  url: 'https://waitrose.com',
  info:
    '(<a href="https://mindera.com" target="_blank">Mindera</a>) Working as FE developer on <a href="https://waitrose.com" target="_blank">Waitroseâ€˜s website</a>.',
  techs: [{ name: 'ReactJS', url: 'https://reactjs.org/' }, { name: 'Redux', url: 'https://redux.js.org/' }],
}

test('Experience Card display correctly', () => {
  let tree = renderWithTheme(<ExperienceCard {...mockProps} />).toJSON()
  expect(tree).toMatchSnapshot()
})
