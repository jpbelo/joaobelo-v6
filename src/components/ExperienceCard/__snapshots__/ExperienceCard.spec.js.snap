// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience Card display correctly 1`] = `
.c2 {
  font-size: 11px;
  display: inline-block;
  border-radius: 6px;
  padding: 0 10px;
  margin-right: 6px;
  margin-bottom: 6px;
  line-height: 2;
  background-color: hsl(0,0%,90%);
  -webkit-transition: background-color 250ms ease-in-out;
  transition: background-color 250ms ease-in-out;
}

.c2[href^='http'] {
  background-color: hsl(46,50%,92%);
}

.c2[href^='http']:hover {
  background: hsl(0,50%,90%);
}

.c1 {
  margin-top: 10px;
}

.c0 {
  margin-bottom: 32px;
}

.c0 h2 {
  margin-bottom: 10px;
}

.c0 h3 {
  font-size: 12px;
  margin-bottom: 4px;
}

.c0 p {
  margin-bottom: 15px;
}

<div
  className="c0"
>
  <h2>
    Waitrose
  </h2>
  <h3>
    2019-04-19
     to 
    present
  </h3>
  <p
    dangerouslySetInnerHTML={
      Object {
        "__html": "(<a href=\\"https://mindera.com\\" target=\\"_blank\\">Mindera</a>) Working as FE developer on <a href=\\"https://waitrose.com\\" target=\\"_blank\\">Waitroseâ€˜s website</a>.",
      }
    }
  />
  <div
    className="c1"
  >
    <a
      className="c2"
      href="https://reactjs.org/"
      target="_blank"
    >
      ReactJS
    </a>
    <a
      className="c2"
      href="https://redux.js.org/"
      target="_blank"
    >
      Redux
    </a>
  </div>
</div>
`;
